test/release/examples/primers/specialMethods.chpl

```diff
@@ -0,0 +1,172 @@
+// Special Methods
+
+/* This primer covers special methods for classes and records */
+
+// Classes and records can have a handful of methods that are specially named
+// and treated differently than other methods.  These methods enable
+// operations such as initializing or deinitializing a class or record
+// instance, accessing a class or record as though it were an array,
+// iterating over an class or record, and defining the way a class or record
+// is read from or written to a channel.
+//
+// Note that there are two ways to declare a method. Methods declared within
+// a class or record are called primary methods:
+
+record ExampleRecord1 {
+  var exampleField: int;
+  proc primaryMethod() { }
+}
+
+
+// Methods declared outside of a class or record are called secondary methods:
+
+record ExampleRecord2 {
+  var exampleField: int;
+}
+proc ExampleRecord2.secondaryMethod() { }
+
+// This primer will use the secondary method form, but all of the special
+// methods can also be written as primary methods.
+
+// First we will declare a simple record with a field that is a tuple of
+// integers.  We'll add special methods and iterators to this record later.
+record R {
+  param size: int = 10;
+  var vals: size*int;
+}
+
+/*
+  Initializers and Deinitializers
+  -------------------------------
+*/
+
+// An initializer is called when creating an instance of the class or record,
+// for example with the ``new`` keyword. An initializer accepting zero
+// arguments is called a *default initializer*.  An initializer that takes
+// another instance of the same class or record is called a *copy initializer*.
+
+// The ``deinit`` method will deinitialize a record when it leaves scope,
+// or a class when ``delete`` is called on it. If the class or record
+// contained any unmanaged classes, open files, etc. this method would be
+// the place to delete them or otherwise clean them up. See the :doc:`records`
+// primer for more details on initializers and deinitializers.
+
+/*
+  The ``this`` Accessor
+  ---------------------
+*/
+
+// The ``this`` method gives the record the ability to be accessed like an
+// array.  Here we use the index to choose a tuple element.
+proc R.this(n: int) ref {
+  if n < 1 || n > size then
+    halt("index out of bounds accessing R");
+  return vals[n];
+}
+
+var r = new R();
+
+// All functions and methods in Chapel can be called using either parenthesis
+// or square brackets. Here we use square brackets to style it as an array
+// access.
+r[1] = 1;
```

Comment:
Looking at the rendered form, I'd probably move the declaration of `r` after the comment so that it's closer to its uses than the definition of the .this method.  I _might_ also change these examples to use square brackets in one case and parens in the other (and remove the rationale for using square from the comment) to illustrate the observation that either can be used rather than suggesting that one or the other would be more appropriate for a given situation (?).